Assets/_BForBoss/_Utility/Scripts/Editor/AttributeValidators/ResolveAttributeValidator.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using Sirenix.OdinInspector.Editor.Validation;
+
+[assembly: RegisterValidator(typeof(Perigon.Utility.ResolveAttributeValidator))]
+
+namespace Perigon.Utility
+{
+    public class ResolveAttributeValidator : AttributeValidator<ResolveAttribute>
+    {
+        protected override void Validate(ValidationResult result)
+        {
+            Type fieldType = Property.Info.TypeOfValue;
+
+            //Check if field type is Struct
```

Comment:
NiT:  if you don't wanna write this comment, you can put whatever is in the if statement into a separate bool 
```suggestion
           var isFieldTypeStruct = fieldType.IsValueType && !fieldType.IsPrimitive && fieldType != typeof(decimal) &&
                fieldType != typeof(DateTime) && !fieldType.IsEnum
```

Reply:
Fair sounds good, I was actually thinking of making an extension method for `Type` class and name it `IsStruct`, instead of this mess. What do you think @m4rtint

Reply:
My at work staff dev sort of made us adopt a rule of 3.

If you found that it's used in 3 different places, then extract it out.

Reply:
Just a tip, but it's ultimately up to you.